### PR TITLE
Fix false calculation of w (omega) when resolveInFrame == frame_resolve

### LINKED ISSUE #4145: Error with frame_resolve math in BasicAbsoluteAngularVelocity sensor
https://github.com/modelica/ModelicaStandardLibrary/blob/a963d5bac1ac2b6c00ea1ca0567036a17f9f5262/Modelica/Mechanics/MultiBody/Sensors/Internal/BasicAbsoluteAngularVelocity.mo#L23

This function resolves the angular velocity of `frame_a.R` in the world frame, which is then resolved from `frame_a` to `frame_resolve`. That is not correct, as it resolves a world frame angular velocity from `frame_a` to `frame_resolve`.

There are two possible updates, with option 2 being preferred because I believe there are less operations involved.
1. `w = Frames.resolveRelative(Frames.angularVelocity2(frame_a.R), frame_a.R, frame_resolve.R);` where `Frames.angularVelocity1(frame_a.R)` is replaced by `Frames.angularVelocity2(frame_a.R)`.
2. `w = Frames.resolve2(frame_resolve.R, Frames.angularVelocity1(frame_a.R));` where the world frame angular velocity of `frame_a` is resolved in `frame_resolve` directly.

### PR DESCRIPTION
**Pull request by tobolar**
Close #4145 

@beutlich Shall this be backported to v3.2.3 as well?
Oh, I see there is https://github.com/modelica/ModelicaStandardLibrary/wiki/Onboarding-MAP-Lib-Project-Leader-and-Deputy#msl-github-issues-and-pull-requests.
Will try to prepare a proper back-port once this PR is reviewed.

Backport to v3.2.3: #4154

### CODE REVIEW

File: ModelicaTest/MultiBody.mo [hunk 1]
```diff
@@ -8844,44 +8844,51 @@ This test model was proposed in ticket <a href=\"https://github.com/modelica/Mod
...
       parameter Real tol=1e-3;
-      inner MultiBody.World world(gravityType=MultiBody.Types.GravityTypes.UniformGravity)
-        annotation (Placement(transformation(extent={{-80,0},{-60,20}})));
+      inner MultiBody.World world(
+        enableAnimation=false,
+        gravityType=MultiBody.Types.GravityTypes.UniformGravity)
+        annotation (Placement(transformation(extent={{-100,-10},{-80,10}})));
       MultiBody.Joints.Revolute rev(
-        n={0,0,1},
+        n(displayUnit="1") ={0,0,1},
```
Comment (HansOlsson) [inline ModelicaTest/MultiBody.mo:8852]: Why change displayUnit for n?

File: ModelicaTest/MultiBody.mo [hunk 2]
```diff
@@ -8844,44 +8844,51 @@ This test model was proposed in ticket <a href=\"https://github.com/modelica/Mod
...
         useAxisFlange=true,
         phi(fixed=true),
-        w(fixed=true)) annotation (Placement(transformation(extent={{-40,0},{-20,
-                20}})));
+        w(fixed=true)) annotation (Placement(transformation(extent={{-60,-10},{-40,10}})));
+      MultiBody.Joints.Revolute rev1(
+        n(displayUnit="1") = {1,0,0},
```
Comment (HansOlsson) [inline ModelicaTest/MultiBody.mo:8857]: (And similarly for this n).

File: ModelicaTest/MultiBody.mo (hunk 1 above)
Comment (tobolar) [reply ModelicaTest/MultiBody.mo:8852]: This was added to `rev1` by Dymola 2023x, so I added this to `rev` as well.

File: ModelicaTest/MultiBody.mo (hunk 2 above)
Comment (tobolar) [reply ModelicaTest/MultiBody.mo:8857]: This was added to by Dymola 2023x. I didn't modify this.

**Review by HansOlsson — APPROVED** [review]
Seems ok.

File: ModelicaTest/MultiBody.mo [hunk 3]
```diff
@@ -8844,44 +8844,51 @@ This test model was proposed in ticket <a href=\"https://github.com/modelica/Mod
     model AbsoluteSensor "Simple pendulum with one revolute joint and one body"
       extends Modelica.Icons.Example;
       parameter Real tol=1e-3;
-      inner MultiBody.World world(gravityType=MultiBody.Types.GravityTypes.UniformGravity)
-        annotation (Placement(transformation(extent={{-80,0},{-60,20}})));
+      inner MultiBody.World world(
+        enableAnimation=false,
```
[the remark this answers is not in the document]
Comment (beutlich) [reply ModelicaTest/MultiBody.mo:8848]: Indeed, would be easier to review otherwise.

[the remark this answers is not in the document]
Comment (tobolar) [reply ModelicaTest/MultiBody.mo]: I have deleted the modifier.

**Review by MartinOtter — APPROVED** [review]

**Comment by beutlich** [comment]
@tobolar Can ou please also create a back-port PR for maint/4.0.x (now, that ist is fixed in master and main/3.2.3). Thank you!

**Pull request merged**